### PR TITLE
nix-daemon: no longer preallocate users and enable uid-range feature …

### PR DESCRIPTION
**Pull request by Mic92**
…for container builds

### CODE REVIEW

**Comment by zimbatm** [comment]
how reliable is this?

**Comment by zowoq** [comment]
These should be opt in, not enabled by default.

File: nixos/common/nix.nix [hunk 1]
```diff
@@ -9,9 +9,17 @@
     "flakes"
     "repl-flake"
     "impure-derivations"
+    # for container in builds support
     "auto-allocate-uids"
+    "cgroup"
   ];
 
+  # no longer need to pre-allocate build users for everything
+  nix.settings.auto-allocate-uids = true;
```
Comment (SuperSandro2000) [inline nixos/common/nix.nix:18]: Can I ask a stupid question: Would I need to set use-cgroups, too? When running nix show-config it showed as being disabled but auto-allocate-uids mentions that it requires this feature and it worked without it for me.

**Comment by Mic92** [comment]
> These should be opt in, not enabled by default.

Please explain. I missing the context here.

**Comment by Mic92** [comment]
> how reliable is this? It has the potential to break a lot of things.

You have to explicitly request this in your derivation: https://github.com/NixOS/nix/blob/955bbe53c5e7e742f7f584c54f54b09fd5ce4009/tests/nixos/containers/systemd-nspawn.nix#L47
Otherwise it's a no-op.

**Comment by zowoq** [comment]
> Regarding `nix.settings.auto-allocate-uids`: This should have been tested in production by the determinate systems installer for a while.

It was disabled last month.

**Comment by phaer** [comment]
@zowoq If we are talking about https://github.com/DeterminateSystems/nix-installer/pull/638 et al: This seems to affect non-nixos linux distros only, while srvos only runs on nixos?

**Comment by zowoq** [comment]
I wasn't talking about why, I was just pointing out that it isn't currently enabled.

**Comment by phaer** [comment]
Okay, so the answer to the question why those settings should not be enabled by default is left as an exercise to the reader? 😄

**Comment by zowoq** [comment]
> Okay, so the answer to the question why those settings should not be enabled by default is left as an exercise to the reader? 😄

I don't understand? I was responding to this:

> This should have been tested in production by the determinate systems installer for a while.

It was is only "tested in production" for a few months.

**Comment by phaer** [comment]
Maybe I misread, sorry for the noise if so. In my reading the conversation had started with 

> these should be opt in, not enabled by default

Which sounded like a response implying specific knowledge about the question in the post before it. Anyway, it's merged so we'll gonna find out 😄

**Comment by Mic92** [comment]
> Maybe I misread, sorry for the noise if so. In my reading the conversation had started with
> 
> > these should be opt in, not enabled by default
> 
> Which sounded like a response implying specific knowledge about the question in the post before it. Anyway, it's merged so we'll gonna find out 😄

It's not enabled by default in srvos. It's a different module that can be enabled optionally.

**Pull request merged**